jax/core.py

```diff
@@ -909,7 +909,9 @@ def __repr__(self):
   _oct     = concretization_function_error(oct)
 
   def at_least_vspace(self) -> AbstractValue:
-    return self
+    # TODO(lenamartens): maybe defensive, where do UnshapedArrays come in?
```

Comment:
This looks good to me!

I think UnshapedArrays never arise nowadays. We haven't removed them because I think they're conceptually nice, but we never made use of them because tracing at an unshaped level is not very useful.